client/lib/widgets/top_nav.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:client/constants/resources.dart';
+import 'package:client/helpers/responsiveness.dart';
+import 'package:client/themes/theme_constants.dart';
+import 'package:flutter/material.dart';
+
+AppBar topNavigationBar(BuildContext context, GlobalKey<ScaffoldState> key) =>
+    AppBar(
+      leading: !Responsive.isMobile(context)
+          ? Row(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.only(left: apppadding),
+                  child: Image.asset(
+                    Resources.icon,
+                    width: 28,
+                  ),
+                ),
+              ],
+            )
+          : IconButton(
+              icon: const Icon(Icons.menu),
+              onPressed: () {
+                key.currentState!.openDrawer();
+              }),
+      title: Row(
+        children: [
+          Visibility(
+            visible: !Responsive.isMobile(context),
+            child: const Text('Open Food Facts'),
+          ),
+          Expanded(child: Container()),
```

Comment:
Why do you need this? A empty `Container` is really expensive

Reply:
![image](https://user-images.githubusercontent.com/96618001/176730169-f52cb6fe-1c69-4e2e-b611-88b2c90eac10.png)